proto/message.py

```diff
@@ -398,18 +407,43 @@ def to_json(
                 Pass None for the most compact representation without newlines.
             float_precision (Optional(int)): If set, use this to specify float field valid digits.
                 Default is None.
+            always_print_fields_with_no_presence (Optional(bool)): If True, fields without
+                presence (implicit presence scalars, repeated fields, and map fields) will
+                always be serialized. Any field that supports presence is not affected by
+                this option (including singular message fields and oneof fields). If
+                `including_default_value_fields` is set to False, this option has no effect.
         Returns:
             str: The json string representation of the protocol buffer.
         """
-        return MessageToJson(
-            cls.pb(instance),
-            use_integers_for_enums=use_integers_for_enums,
-            including_default_value_fields=including_default_value_fields,
-            preserving_proto_field_name=preserving_proto_field_name,
-            sort_keys=sort_keys,
-            indent=indent,
-            float_precision=float_precision,
-        )
+        # For backwards compatibility of this breaking change in protobuf 5.x which is specific to proto2
+        # https://github.com/protocolbuffers/protobuf/commit/26995798757fbfef5cf6648610848e389db1fecf
+        if PROTOBUF_VERSION[0] in ("3", "4"):
+            return MessageToJson(
+                cls.pb(instance),
+                use_integers_for_enums=use_integers_for_enums,
+                including_default_value_fields=including_default_value_fields,
```

Comment:
If we're deprecating this field on thisA PI surface, we should have the same deprecation logic on the RHS here as in the `else` branch (and I suggest the `print_field` variable I defined in my other comment).

Reply:
Fixed in https://github.com/googleapis/proto-plus-python/pull/433/commits/0e65d18eefc21ef7b34b9fc12f9aa779affca437